### PR TITLE
Imply MODULARIZE when EXPORT_ES6 is set

### PR DESCRIPTION
**Pull request by algestam**
Fix for issue #12525

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1641,7 +1641,8 @@ def include_and_export(name):
       exit_with_error('If pthreads and memory growth are enabled, MAXIMUM_MEMORY must be set')
 
     if shared.Settings.EXPORT_ES6 and not shared.Settings.MODULARIZE:
-      exit_with_error('EXPORT_ES6 requires MODULARIZE to be set')
+      # EXPORT_ES6 requires output to be a module
+      shared.Settings.MODULARIZE = 1
```
Comment (RReverser) [inline emcc.py:1645]: Is there a way we could still throw an error in case user explicitly set `-s MODULARIZE=0` but not when they omitted the option?

Comment (RReverser) [reply emcc.py:1645]: Great. Let's do that, @algestam.

Comment (algestam) [reply emcc.py:1645]: Thanks, will implement this and update the PR.

Comment (RReverser) [reply emcc.py:1647]: Looks great, thanks! It's odd there isn't an existing test for the previous behaviour, but perhaps worth adding one (to test couple of scenarios - no `MODULARIZE` is provided, explicit `MODULARIZE=1` is provided and explicit `MODULARIZE=0` is provided)?

Comment (curiousdannii) [reply emcc.py:1647]: There's no tests for EXPORT_ES6 at all... it's a big blind spot.

Comment (sbc100) [reply emcc.py:1647]: `EXPORT_ES6` is automatically set when `.mjs` is uses as the output filename so we do have a tiny bit of testing :)

```
$ git grep "\.mjs" tests/*.py
tests/test_other.py:    self.run_process([EMCC, '-o', 'hello_world.mjs', path_from_root('tests', 'hello_world.c')])
tests/test_other.py:    with open('hello_world.mjs') as f:
tests/test_other.py:    # self.assertContained('hello, world!', self.run_js('hello_world.mjs'))
tests/test_other.py:    self.run_process([EMCC, '-o', 'hello_world.mjs', '-pthread', '-O1',
tests/test_other.py:    with open('hello_world.mjs') as f:
```

Comment (algestam) [reply emcc.py:1647]: I can look into adding tests that checks that:
* `MODULARIZE` also is set when `EXPORT_ES6` is set
* An error is raised if `EXPORT_ES6` is set and `MODULARIZE` is explicitly disabled

Is `tests/test_other.py` a good place to add these tests?

Comment (RReverser) [reply emcc.py:1647]: @algestam Yeah, that sounds good. Probably add them somewhere right after existing `mjs` test.

Comment (algestam) [reply emcc.py:1647]: Thanks, will give it a try and update the PR.

**Review by RReverser — APPROVED** [review]
Looks good to me, thanks @algestam!

**Comment by RReverser** [comment]
@sbc100 @kripken Any objection to me merging this?

File: tests/test_other.py [hunk 2]
```diff
@@ -199,6 +199,15 @@ def test_emcc_output_worker_mjs(self):
     with open('hello_world.worker.js') as f:
       self.assertContained('import(', f.read())
 
+  def test_export_es6_implies_modularize(self):
+    self.run_process([EMCC, path_from_root('tests', 'hello_world.c'), '-s', 'EXPORT_ES6=1', '-s', 'ASSERTIONS=0'])
```
Comment (sbc100) [inline tests/test_other.py:203]: Do you need `ASSERTIONS=0` here?

File: tests/test_other.py [hunk 3]
```diff
@@ -199,6 +199,15 @@ def test_emcc_output_worker_mjs(self):
...
+    src = open('a.out.js').read()
+    self.assertContained('export default Module;', src)
+
+  def test_export_es6_requires_modularize(self):
+    err = self.expect_fail([EMCC, path_from_root('tests', 'hello_world.c'), '-s', 'EXPORT_ES6=1', '-s', 'MODULARIZE=0', '-s', 'ASSERTIONS=0'])
```
Comment (sbc100) [inline tests/test_other.py:208]: and here?

**Review by sbc100 — APPROVED** [review]
lgtm % nits

File: tests/test_other.py (hunk 3 above)
Comment (algestam) [reply tests/test_other.py]: Good catch, those are not needed. Have updated the PR. Thanks!

File: tests/test_other.py [hunk 4]
```diff
@@ -199,6 +199,15 @@ def test_emcc_output_worker_mjs(self):
     with open('hello_world.worker.js') as f:
       self.assertContained('import(', f.read())
 
+  def test_export_es6_implies_modularize(self):
+    self.run_process([EMCC, path_from_root('tests', 'hello_world.c'), '-s', 'EXPORT_ES6=1'])
+    src = open('a.out.js').read()
```
Comment (RReverser) [inline tests/test_other.py:204]: I think you should be using `with open(...)` pattern like in other tests (see the one right above) to ensure that file is properly closed.

Comment (algestam) [reply tests/test_other.py]: Thanks, that's a better way. The PR has been updated :thumbsup:

**Review by kripken — APPROVED** [review]
Great!

**Pull request merged**